fl4health/clients/basic_client.py

```diff
@@ -456,6 +456,11 @@ def train_by_epochs(
             self.train_loss_meter.clear()
             for input, target in self.train_loader:
                 input, target = input.to(self.device), target.to(self.device)
+                # Assume first dimension is batch size. Sampling iterators (such as Poisson batch sampling), can
+                # construct empty batches. We skip the iteration if this occurs.
+                if len(input) == 0:
```

Comment:
Would it be better if we check for this before moving `input` and `target` to the GPU?

Reply:
Yes it would!